printer/printer.go

```diff
@@ -74,21 +101,85 @@ func Print(v interface{}) {
 }
 
 // Flush writes the elements accumulated in the printer
-func Flush() {
-	if printer.Format == FormatJSON {
-		var b []byte
-		switch {
-		case printer.Single && len(printer.Lines) == 0:
-			return
-		case printer.Single && len(printer.Lines) == 1:
-			b, _ = json.MarshalIndent(printer.Lines[0], "", "  ")
-		default:
-			b, _ = json.MarshalIndent(printer.Lines, "", "  ")
+func Flush() error {
+	opts := printOpts{
+		format: printer.Format,
+		single: printer.Single,
+	}
+
+	cmd := printer.cmd
+	if cmd != nil {
+		shortStat, err := printer.cmd.Flags().GetBool("short-stat")
+		if err == nil && printer.cmd.Name() == "list" && printer.cmd.Parent().Name() != "auth" {
```

Comment:
Relying on command names feels a bit brittle. Maybe not for this PR, but wondering if this is the time to start thinking about wrapping `cobra.Command` into our own struct and adding this kind of information there.

Reply:
I was thinking the same actually but we can put these into command's Context maybe.

Reply:
I've checked how context is modified and it looks like the only way to set it is through the `ExecuteContext` calls, which maybe makes it a bit more difficult, but `cmd.Context()` is a good option if it's doable

Reply:
Yeah checked and it works quite different then anticipated :(